### PR TITLE
Updated by adding controller ,  get by pagination and search function

### PR DESCRIPTION
**Pull request by Michaeljogoh**
Add query pagination on get method and search middleware, for users to search for  blog posts

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -1,10 +1,17 @@
 const express = require('express');
 const app = express();
-const dotenv = require('dotenv').config();
+require('dotenv').config();
 const cors = require('cors');
 const blogRoutes = require('./routes/blogRoutes');
+const mongoose = require('mongoose');
+const { connect } = require('http2');
```
Comment (Asylcreek) [inline app.js:7]: What did you use this for?

File: controller/blogController.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+const blogPost = require('../model/blogModel');
+const router = require('../routes/blogRoutes');
```
Comment (Asylcreek) [inline controller/blogController.js:2]: Please remove this import

File: controller/blogController.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+const blogPost = require('../model/blogModel');
```
Comment (Asylcreek) [inline controller/blogController.js:1]: All your models should be PascalCase, in this case 'BlogModel'

File: controller/blogController.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+const router = require('../routes/blogRoutes');
+
+// post 
+const postBlog = async (req , res) =>{
+    const {title , content , author } = req.body
+    const postBlog = new blogPost({title , content , author});
+    await postBlog.save();
```
Comment (Asylcreek) [inline controller/blogController.js:8]: Look up `Model.create` and use that pattern instead

File: controller/blogController.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+const count = await blogPost.countDocuments();
+res.json({getPosts, totalPages:Math.ceil(count / limit), currentPage: page})
+};
+
+// search
+
+const searchPostBlogs =  async (req , res)=>{
+    const searchPostBlogs = await blogPost.find({'$or':[{title:{$regex:req.params.word.toLowerCase()}}]})
+    res.status(200).json({searchPostBlogs})
+}
```
Comment (Asylcreek) [inline controller/blogController.js:30]: Move this searching functionality inside the `getPostBlog` function; whenever you search, you would also like to paginate the results

File: controller/blogController.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+    .exec();
+    //Get Total documents in blogPost collection
+const count = await blogPost.countDocuments();
+res.json({getPosts, totalPages:Math.ceil(count / limit), currentPage: page})
+};
+
+// search
+
+const searchPostBlogs =  async (req , res)=>{
+    const searchPostBlogs = await blogPost.find({'$or':[{title:{$regex:req.params.word.toLowerCase()}}]})
```
Comment (Asylcreek) [inline controller/blogController.js:28]: 1. Why the `$or` operator?
2. Look up the `$options` operator to replace the `req.params.word.toLowerCase` call

**Review by Asylcreek — CHANGES_REQUESTED** [review]
Everytime you push, I should not be able to see your `.env` file, always add it to your `.gitignore`

File: controller/blogController.js (hunk 3 above)
Comment (Michaeljogoh) [reply controller/blogController.js:1]: ok i will change it.

File: controller/blogController.js (hunk 5 above)
Comment (Michaeljogoh) [reply controller/blogController.js:30]: ok sir

File: controller/blogController.js (hunk 6 above)
Comment (Michaeljogoh) [reply controller/blogController.js:28]: I will  do that

**Pull request merged**